docs/features/nullable-reference-types.md

```diff
@@ -0,0 +1,139 @@
+List<object?>? OptListOptElements; // list may be null, elements may be null
+```
+In metadata, nullable reference types are annotated with a `[Nullable]` attribute.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(
+        AttributeTargets.Module |
+        AttributeTargets.Class |
+        AttributeTargets.Event | AttributeTargets.Field | AttributeTargets.GenericParameter | AttributeTargets.Property |
+        AttributeTargets.Parameter | AttributeTargets.ReturnValue,
+        AllowMultiple = false)]
+    public sealed class NullableAttribute : Attribute
+    {
+        public NullableAttribute() { }
+        public NullableAttribute(bool[] b) { }
+    }
+}
+```
+If the feature is enabled, a `[module: Nullable]` attribute is emitted to the module.
+```c#
+// C# representation of metadata
+[module: Nullable]
+[Nullable] string OptString; // string?
+[Nullable(new[] { true, true })] List<object> OptListOptElements; // List<object?>?
+```
+The `NullableAttribute` type declaration is synthesized by the compiler if it is not included in the compilation.
+
+Unannotated reference types are non-nullable or null-oblivious depending whether the containing scope includes `[NonNullTypes]`.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(
+        AttributeTargets.All,
+        AllowMultiple = false)]
+    public sealed class NonNullTypesAttribute : Attribute
+    {
+        public NonNullTypesAttribute(bool enabled = true) { }
+    }
+}
+```
+If there is no `[NonNullTypes]` attribute in any containing scope, including the module, reference types are non-nullable.
+```c#
+[NonNullTypes(false)] string[] Oblivious; // string~[]~
+[NonNullTypes(true)] string[] NotNull; // string![]!
+[NonNullTypes(false), Nullable(new[] { false, true })] string[] ObliviousMaybeNull; // string?[]~
+[NonNullTypes(true), Nullable(new[] { false, true })] string[] NotNullMaybeNull; // string?[]!
+```
+`NonNullTypesAttribute` is not synthesized by the compiler. If the attribute is used explicitly in source, the type declaration must be provided explicitly to the compilation. The type should be defined in the framework.
+
+## Declaration warnings
+_Describe warnings reported for declarations in initial binding._
+
+## Flow analysis
+Flow analysis is used to infer the nullability of variables within executable code. The inferred nullability of a variable is independent of the variable's declared nullability.
+
+### Warnings
+_Describe set of warnings. Differentiate W warnings._
+
+### Null tests
+_Describe the set of tests that affect flow state._
+
+### Conversions
+_Describe valid top-level and variance conversions._
+_Describe warnings from user-defined conversions._
+
+### Assignment
+For `x = y`, the nullability of the converted type of `y` is used for `x`.
+Warnings are reported for top-level and nested nullability mismatches.
+The warning assigning `?` to `!` is a W warning if the target is a local.
```

Comment:
>if the target is a local. [](start = 48, length = 25)

Is this correct? 
I think there are other cases, such as assigning to a parameter, to a field, or to a ref-returning method call.